src/screen/ProductPage.tsx

```diff
@@ -114,65 +156,107 @@ export function ProductPage(props: ProductPageProps) {
-                    value="Delete"
-                    onClick={() =>
-                      delProduct({
-                        id: product.id,
-                        title: product.title,
-                        price: product.price,
-                        categoryId: product.categoryId,
-                      }).then(() => {
-                        reFetchProduct();
-                      })
-                    }
-                  ></input>
-                  <input
-                    type="button"
-                    value="Edit"
-                    onClick={() => {
-                      setEditProduct(product.id);
-                      setInputType("edit");
-                      setInputProduct({
-                        id: product.id,
-                        title: product.title,
-                        price: product.price,
-                        categoryId: product.categoryId,
-                      });
-                    }}
-                  ></input>
-                </td>
-              </tr>
-            );
-          })}
-        </tbody>
-      </table>
+      {(function () {
+        if (fetchProduct.isLoading || fetchCategories.isLoading) {
+          return <p>Data is Loading</p>;
+        } else if (
+          createProduct.errorMessage ||
+          deleteProduct.errorMessage ||
+          editProduct.errorMessage ||
+          fetchProduct.errorMassage ||
+          fetchCategories.errorMessage
+        ) {
+          return (
+            <p>
+              {createProduct.errorMessage ||
+                deleteProduct.errorMessage ||
+                editProduct.errorMessage ||
+                fetchProduct.errorMassage ||
+                fetchCategories.errorMessage}
+            </p>
+          );
+        } else if (fetchProduct.products.length === 0) {
+          return <p>Data is Empty</p>;
+        } else {
+          return (
+            <table>
+              <thead>
+                <tr>
+                  <th>ID Product</th>
+                  <th>Title</th>
+                  <th>Price</th>
+                  <th>Category</th>
+                  <th>Action</th>
+                </tr>
+              </thead>
+              <tbody>
+                {fetchProduct.products.map((product) => {
+                  return (
+                    <tr key={product.id}>
+                      <td>{product.id}</td>
+                      <td>{product.title}</td>
+                      <td>{product.price}</td>
+                      <td>{categoryTitle(product.categoryId)}</td>
+                      <td>
+                        <input
+                          type="button"
+                          value="Delete"
+                          disabled={
+                            deleteProduct.isLoading
+                              ? true
+                              : inputType === "edit"
+                              ? true
+                              : false
+                          }
```

Comment:
Ini bisa disingkat jadi kayak gini

```suggestion
                          disabled={deleteProduct.isLoading || inputType === "edit"}
```

Reply:
updated